vnext/Shared/Utils.cpp

```diff
@@ -4,10 +4,49 @@
 #include "Utils.h"
 #include <regex>
 
+#include <DesktopWindowBridge.h>
+#include <ShlObj.h>
+#include <Shlwapi.h>
+#include <appmodel.h>
+#include <winrt/Windows.Storage.h>
+
 using std::string;
 
 namespace Microsoft::React {
 
+std::future<std::string> getApplicationDataPath(const wchar_t *childFolder) {
+  uint32_t length{0};
+  bool isPackagedApp = GetCurrentPackageFullName(&length, nullptr) != APPMODEL_ERROR_NO_PACKAGE;
+  std::string appDataPath;
+  if (isPackagedApp) {
+    if (!childFolder) {
+      co_return winrt::to_string(winrt::Windows::Storage::ApplicationData::Current().LocalFolder().Path());
+    } else {
+      auto subfolder = co_await winrt::Windows::Storage::ApplicationData::Current().LocalFolder().CreateFolderAsync(
+          childFolder, winrt::Windows::Storage::CreationCollisionOption::OpenIfExists);
+      co_return winrt::to_string(subfolder.Path());
+    }
+  } else {
+    wchar_t wzAppData[MAX_PATH]{};
```

Comment:
Question: Now that windows is no longer hampered by maxPath, is this still the right thing to do?

Reply:
In trying to answer my own question I noticed that this Api is deprecated:
https://docs.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetfolderpatha?redirectedfrom=MSDN

Reply:
And the replacement function returns the pointer: [`SHGetKnownFolderPath`](https://docs.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetknownfolderpath)

Reply:
Thanks !